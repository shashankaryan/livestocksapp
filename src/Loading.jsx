// @flow

import React from 'react'
import styled, { keyframes } from 'styled-components'
import image from './static/img/loading.png'

const spin = keyframes`
  from{
    transform: rotate(0deg);
  }
  to{
    transform: rotate(360deg);
  }
`

const Image = styled.img`
	animation: ${spin} 2s infinite linear;
	margin-top: 10px;
	height: 30px;
	width: 30px;
`

const Loading = () => <Image src={image} alt="loading indicator" />

export default Loading
