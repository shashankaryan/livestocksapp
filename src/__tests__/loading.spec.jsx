import React from 'react'
import { shallow } from 'enzyme' // render, static is also available. but they are slower.
import Loading from '../Loading'

test('Table Item renders correctly', () => {
	const component = shallow(<Loading />)
	expect(component).toMatchSnapshot()
})
