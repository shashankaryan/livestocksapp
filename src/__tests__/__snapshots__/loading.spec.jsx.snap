// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Item renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.img
    alt="loading indicator"
    src="/public/img/loading.png"
/>,
  "nodes": Array [
    <styled.img
      alt="loading indicator"
      src="/public/img/loading.png"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Loading />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.img
          alt="loading indicator"
          src="/public/img/loading.png"
/>,
        "_debugID": 2,
        "_renderedOutput": <styled.img
          alt="loading indicator"
          src="/public/img/loading.png"
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Loading />,
}
`;
